src/vmspawn/vmspawn.c

```diff
@@ -0,0 +1,480 @@
+
+static int run_virtual_machine(void) {
+        int r;
+        const char *accel = "tcg";
+        _cleanup_strv_free_ char **cmdline = NULL;
+
+        bool auto_conf_kvm = arg_qemu_kvm == CONFIG_FEATURE_AUTO && qemu_check_kvm_support();
+        if (arg_qemu_kvm == CONFIG_FEATURE_ENABLED || auto_conf_kvm)
+                accel = "kvm";
+
+        const char *ovmf = NULL;
+        r = find_ovmf_firmware(&ovmf);
+        if (r < 0)
+                return log_error_errno(r, "Couldn't find OVMF UEFI firmware blob.");
+        bool ovmf_supports_sb = r == 0;
+        if (!ovmf_supports_sb)
+                log_warning("Couldn't find OVMF firmware blob with secure boot support, "
+                            "falling back to OVMF firmware blobs without secure boot support.");
+        const char *smm = ovmf_supports_sb ? "on" : "off";
+
+        _cleanup_free_ char *machine = NULL;
+#ifdef __aarch64__
+        machine = strjoin("type=virt,accel=", accel);
+#else
+        machine = strjoin("type=q35,accel=", accel, ",smm=", smm);
+#endif
+        if (!machine)
+                return log_oom();
+
+        _cleanup_free_ char *qemu_binary = NULL;
+        r = find_qemu_binary(&qemu_binary);
+        if (r == -ESRCH)
+                return log_error_errno(r, "Native architecture is not supported by qemu.");
+        if (r < 0)
+                return r;
+
+        const char *smp = arg_qemu_smp ? arg_qemu_smp : "1";
+        const char *mem = FORMAT_BYTES(arg_qemu_mem);
+        cmdline = strv_new(
+                qemu_binary,
+                "-machine", machine,
+                "-smp", smp,
+                "-m", mem,
+                "-object", "rng-random,filename=/dev/urandom,id=rng0",
+                "-device", "virtio-rng-pci,rng=rng0,id=rng-device0",
+                "-nic", "user,model=virtio-net-pci",
+                "-cpu", "max"
+        );
+
+        if (arg_qemu_gui) {
+                r = strv_extend_strv(&cmdline, STRV_MAKE("-vga", "virtio"), false);
+                if (r < 0)
+                        log_oom();
+        } else {
+                r = strv_extend_strv(&cmdline, STRV_MAKE(
+                        "-nographic",
+                        "-nodefaults",
+                        "-chardev", "stdio,mux=on,id=console,signal=off",
+                        "-serial", "chardev:console",
+                        "-mon", "console"
+                ), false);
+                if (r < 0)
+                        log_oom();
+        }
+
+#ifdef ARCHITECTURE_SUPPORTS_SMBIOS
+        ssize_t n;
+        for (size_t i = 0; i < arg_n_credentials; i++) {
+                _cleanup_free_ char *cred_data_b64 = NULL;
+                Credential* cred = &arg_credentials[i];
+
+                n = base64mem(cred->data, cred->size, &cred_data_b64);
+                if (n < 0)
+                        return log_oom();
+
+                r = strv_extend(&cmdline, "-smbios");
+                if (r < 0)
+                        log_oom();
```

Comment:
here and everywhere else: must be `return log_oom()`

Reply:
Sorry I will go and find all of these.

Reply:
Have fixed and squashed them into the commit they came from.